Implement tdi_port_manager.cc

ffoulkes opened the pull request:
- Converted TdiPortManager from an abstract class to a base class.
- Moved common definitions from subclasses to the base class.
- Eliminated vestigial Es2kPortManager methods.

ffoulkes committed:
Implement tdi_port_manager.cc

- Converted TdiPortManager from an abstract class to a base class.
  Revised subclasses to remove common definitions that are now
  in the base class.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Eliminate vestigial Es2kPortManager methods

- Deleted the SetDeflectOnDropDestination, GetPcieCpuPort, and
  SetTmCpuPort methods from Es2kPortManager.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Format tdi_port_manager.cc

Signed-off-by: Derek G Foster <derek.foster@intel.com>

5abeel commented on stratum/hal/lib/tdi/tdi_port_manager.h:
```diff
@@ -70,6 +71,26 @@ class TdiPortManager {
+  static constexpr absl::Duration kWriteTimeout = absl::InfiniteDuration();
+
+  // Default MTU for ports on ES2K.
+  static constexpr int32 kBfDefaultMtu = 10 * 1024;  // 10K
```
Hopefully these dont need to be change based on different targets. But we could possibly override in future in derived classes if needed.

ffoulkes replied:
Good catch. On closer examination, it turns out that the only target that uses this constant is `tofino`.

I've moved the definition from `TdiPortManager` to `TofinoPortManager`.

5abeel commented on stratum/hal/lib/tdi/dpdk/dpdk_port_manager.cc:
```diff
@@ -86,19 +80,6 @@ ::util::Status DpdkPortManager::GetPortCounters(int device, int port,
   return ::util::OkStatus();
 }
 
-::util::Status DpdkPortManager::RegisterPortStatusEventWriter(
```
Do the relevent MOCK routines need to be cleaned up in the dpdk_port_manager_mock.h?

5abeel replied:
Same comment for es2k and tofino targets

ffoulkes replied:
I don't think so. `Es2kPortManagerMock` extends `Es2kPortManager`. It mocks all the methods the parent class supports, including the ones it inherits from `TdiPortManager`. The same is true of the other PortManagerMocks. I believe this is the correct way to define the mock.

5abeel reviewed (approved):
Couple of comments - please verify if more cleanup needed in mock files. Rest LGTM

ffoulkes committed:
Move definition of kBfDefaultMtu

- Moved the definition of kBfDefaultMtu from TdiPortManager to
  TofinoPortManager.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Merge branch 'split-arch' into tdi-port-manager

ffoulkes committed:
Make kBfDefaultMtu private

Signed-off-by: Derek G Foster <derek.foster@intel.com>

nupuruttarwar reviewed (approved):
LGTM
